pkg/machine/shim/host.go

```diff
@@ -245,9 +248,11 @@ func Init(opts machineDefine.InitOptions, mp vmconfigs.VMProvider) error {
 		return err
 	}
 
-	err = ignBuilder.Build()
-	if err != nil {
-		return err
+	if len(opts.IgnitionPath) == 0 {
+		err = ignBuilder.Build()
```

Comment:
your choice but I think that the if err := ignBuilder.Build(); err != nil { syntax is cleaner here? nit

Reply:
Whose choice are you referring to? I'm happy to change it, I'm just unsure if this is an internal discussion between you and @Luap99, or if I should change it.